File: C#(MAIN WORK)/src/OrderDomain/Entities/Order.cs
using System;
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using OrderDomain.ValueObjects;

namespace OrderDomain.Entities
{
    public class Order
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }
        public string CustomerId { get; set; }
        public int Quantity { get; set; } //This will be set auto
        public double Total { get; set; } //This will be set auto
        public string Status { get; set; }
        public Address Address { get; set; }
        public List<string> ProductIds { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
    }
}

File: GO(SECONDARY WORK)/src/Services/OrderService/types/Order.go
package types

import "time"

type Order struct {
	Id         string    `bson:"_id" json:"_id" validate:"isdefault"`              //calculate auto
	CustomerId string    `bson:"customerId" json:"customerId" validate:"required"` //from user
	Quantity   int       `bson:"quantity" json:"quantity" validate:"isdefault"`    //calculate auto
	Total      float32   `bson:"total" json:"total" validate:"isdefault"`          //calculate auto
	Status     string    `bson:"status" json:"status" validate:"required"`         //from user
	Address    Address   `bson:"address" json:"address" validate:"isdefault"`      //get from customer
	ProductIds []string  `bson:"productIds" json:"productIds" validate:"required"` //from user
	CreatedAt  time.Time `bson:"createdAt" json:"createdAt" validate:"isdefault"`  //calculate auto
	UpdatedAt  time.Time `bson:"updatedAt" json:"updatedAt" validate:"isdefault"`  //calculate auto
}
